src/mpid/ch3/channels/nemesis/netmod/tcp/tcp_getip.c

```diff
@@ -67,17 +67,21 @@ int MPIDI_GetIPInterface( MPIDI_CH3I_nem_tcp_ifaddr_t *ifaddr, int *found )
     MPIDI_CH3I_nem_tcp_ifaddr_t myifaddr;
     int nfound = 0, foundLocalhost = 0;
     /* We predefine the LSB and MSB localhost addresses */
-    unsigned int localhost = 0x0100007f;
+	unsigned char localhost6[16] = "::1";
+	unsigned int localhost = 0x0100007f;
 #ifdef WORDS_BIGENDIAN
     unsigned int MSBlocalhost = 0x7f000001;
 #endif
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_GETIPINTERFACE);
 
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_GETIPINTERFACE);
     *found = 0;
-
-    fd = socket(AF_INET, SOCK_DGRAM, 0);
-    if (fd < 0) {
+	/*IPV6
+	if (ipv6_flag_set) 
+	fd = socket(AF_INET6, SOCK_DGRAM, 0);
+    else */
+	fd = socket(AF_INET, SOCK_DGRAM, 0);
```

Comment:
Do you have any idea on how we can switch between opening a IPV6 socket and IPV4 socket?

Reply:
Can we use getifaddr instead of ioctl to get the interface IP address ? 
If we can do that we don't need to choose between a IPV4 and IPV6 socket. 
I just did some research, ioctl is more portable on various OS including linux, unix, windows, aix, solaris, while getifaddr doesn't to be supported in aix and solaris.

Reply:
To your first question, the flag should have been set during the initialization, see comments for `tcp_init.c`.

To your second question, we have to use `ioctl` since we need to support Solaris.